theme/static/js/hs_file_browser.js

```diff
@@ -1080,19 +1080,14 @@ $(document).ready(function () {
 
     function onPaste() {
         var folderName = $("#fb-files-container li.ui-selected").children(".fb-file-name").text();
-        var targetPath = $("#hs-file-browser").attr("data-current-path");
-
-        if (folderName && folderName.lastIndexOf(".") == -1) {  // Makes sure the destination is a folder
```

Comment:
@Maurier Note that it is the backend calls that currently allow erroneous states. Only `move-to-folder` has been updated. The others have insufficient error checking to prevent all problems. Since `move-to-folder` is not used for the other operations, they remain brittle.